go/libraries/doltcore/diff/diff_summary.go

```diff
@@ -195,3 +274,45 @@ func reportKeylessChanges(ctx context.Context, change *diff.Difference, ch chan<
 		return ctx.Err()
 	}
 }
+
+// MapSchemaBasedOnName can be used to map column values from one schema to
+// another schema. A column in |inSch| is mapped to |outSch| if they share the
+// same name and primary key membership status. It returns ordinal mappings that
+// can be use to map key, value val.Tuple's of schema |inSch| to a sql.Row of
+// |outSch|. The first ordinal map is for keys, and the second is for values. If
+// a column of |inSch| is missing in |outSch| then that column's index in the
+// ordinal map holds -1.
+func MapSchemaBasedOnName(inSch, outSch schema.Schema) (val.OrdinalMapping, val.OrdinalMapping, error) {
```

Comment:
Seems like the kind of thing that should have a unit test with a bunch of interesting test cases

Reply:
Added a TODO for this.